Core/GameEngine/Include/GameNetwork/NetPacketStructs.h

```diff
@@ -0,0 +1,405 @@
+// Disconnect Keep Alive Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+// Disconnect keep alive command packet structure
+// Fields: T + type, R + relay, P + playerID, D
+struct NetPacketDisconnectKeepAliveCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Disconnect Player Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+// Disconnect player command packet structure
+// Fields: T + type, R + relay, P + playerID, C + commandID, D + slot + disconnectFrame
+struct NetPacketDisconnectPlayerCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+	UnsignedByte slot;
+	UnsignedInt disconnectFrame;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Packet Router Command Packets
+////////////////////////////////////////////////////////////////////////////////
+
+// Packet router query command packet
+// Fields: T + type, R + relay, P + playerID, D
+struct NetPacketRouterQueryCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+// Packet router ack command packet
+// Fields: T + type, R + relay, P + playerID, D
+struct NetPacketRouterAckCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Disconnect Vote Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+// Disconnect vote command
+// Fields: T + type, R + relay, P + playerID, C + commandID, D + slot + voteFrame
+struct NetPacketDisconnectVoteCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+	UnsignedByte slot;
+	UnsignedInt voteFrame;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Packed Structs for getPackedByteCount() calculations
+// These structs represent the fixed portion of variable-length command messages
+////////////////////////////////////////////////////////////////////////////////
+
+struct NetPacketChatCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+	NetPacketFrameField frame;
+	NetPacketCommandIdField commandId;
```

Comment:
`commandId` comes after `dataHeader`. Is this correct? It is strange that sometimes it comes before and sometimes after. But maybe its is right.